File: Convert/C#/CourseMst_DA.cs
using System;
using System.Collections;
using System.Text;
using Microsoft.VisualBasic.CompilerServices; 
/// <summary>
/// インターネット予約トランザクション管理）のDAクラス
/// <remarks>
/// Author:2018/10/26//QuangTD
/// </remarks>
/// </summary>
public partial class CourseMst_DA : DataAccessorBase
{

    #region  定数／変数 
    private TCourseMstEntity clsTCourseMstEntity = new TCourseMstEntity();
    #endregion

    /// <summary>
    /// 検索用SELECT
    /// </summary>
    /// <param name="paramInfoList"></param>
    /// <remarks></remarks>
    public DataTable getCrsCdExist(Hashtable paramInfoList)
    {
        // 戻り値
        DataTable returnValue = default;
        var sqlString = new StringBuilder();
        {
            var withBlock = clsTCourseMstEntity;
            // SELECT
            sqlString.AppendLine("SELECT");
            sqlString.AppendLine("    CRS_CD ");
            sqlString.AppendLine("FROM");
            sqlString.AppendLine("    T_COURSE_MST ");
            sqlString.AppendLine("WHERE");
            if (!string.IsNullOrEmpty(Conversions.ToString(paramInfoList["crsCd"])))
            {
                sqlString.AppendLine("    T_COURSE_MST.CRS_CD = " + setParam("crsCd", Conversions.ToString(paramInfoList["crsCd"]), withBlock.crsCd.DBType, withBlock.crsCd.IntegerBu, withBlock.crsCd.DecimalBu));
            }
        }

        try
        {
            returnValue = getDataTable(sqlString.ToString());
        }
        catch (Exception ex)
        {
            throw;
        }

        return returnValue;
    }
}

File: Convert/VB/CourseMst_DA.vb
Imports System.Text
''' <summary>
''' インターネット予約トランザクション管理）のDAクラス
''' <remarks>
''' Author:2018/10/26//QuangTD
''' </remarks>
''' </summary>
Public Class CourseMst_DA
    Inherits DataAccessorBase

#Region " 定数／変数 "
    Private clsTCourseMstEntity As New TCourseMstEntity()
#End Region

    ''' <summary>
    ''' 検索用SELECT
    ''' </summary>
    ''' <param name="paramInfoList"></param>
    ''' <remarks></remarks>
    Public Function getCrsCdExist(paramInfoList As Hashtable) As DataTable
        '戻り値
        Dim returnValue As DataTable = Nothing
        Dim sqlString As New StringBuilder
        With clsTCourseMstEntity
            'SELECT
            sqlString.AppendLine("SELECT")
            sqlString.AppendLine("    CRS_CD ")
            sqlString.AppendLine("FROM")
            sqlString.AppendLine("    T_COURSE_MST ")
            sqlString.AppendLine("WHERE")
            If Not String.IsNullOrEmpty(CType(paramInfoList.Item("crsCd"), String)) Then
                sqlString.AppendLine("    T_COURSE_MST.CRS_CD = " & setParam("crsCd", CType(paramInfoList.Item("crsCd"), String), .crsCd.DBType, .crsCd.IntegerBu, .crsCd.DecimalBu))
            End If
        End With
        Try
            returnValue = getDataTable(sqlString.ToString)
        Catch ex As Exception
            Throw
        End Try
        Return returnValue
    End Function
End Class
